src/core.cpp

```diff
@@ -3,55 +3,112 @@
 #include <random>
 
 #include "core.h"
+#include "action.h"
 
-double Character::GetMoney() {
-    return this->money_;
+RelationshipModifier::RelationshipModifier(Character* target, int start_date, int end_date, double opinion_modifier_) {
+    this->target_ = target;
+    this->start_date_ = start_date;
+    this->end_date_ = end_date;
+    this->opinion_modifier_ = opinion_modifier_;
 }
 
-void Character::SetMoney(double money) {
+Character::Character(int id, double money) {
+    this->id_ = id;
     this->money_ = money;
 }
 
-
-Action::Action(Character* actor, double score) {
-    this->actor_ = actor;
-    this->score_ = score;
+int Character::GetId() const {
+    return this->id_;
 }
 
-Action::~Action() {
-}
-
-Character* Action::GetActor() {
-    return this->actor_;
+double Character::GetMoney() const {
+    return this->money_;
```

Comment:
I normally would see `return money_` as opposed to `return this->money_` -- particularly as you're using the trailing underscores so it is pretty clear what is a member variable with using the `this`.

Reply:
seems reasonable. this is another style holdover from Java where you don't typically see underscores like that and instead use "this.foo" to be explicit like that.